### ISSUE TITLE
Hard line break in the editor to make code more readable

### ISSUE THREAD

**Issue by ASalvail** [sharelatex/sharelatex#323]
Something I miss from Emacs Auctex extensions is the capacity to set a specific line length and
have the editor automatically insert a line break each time a word is written that goes over that limit.

Soft line break is achievable by resizing the editor window size, but this method has several problem : 
- A soft line break does not increase the line number;
- The text is dependant on the window size and thus not displayed uniformly on different screens;
- Makes error finding (from line number) that much harder.
This feature would make for more readable LaTeX documents.

To implement this : probably add an option in the settings to specify the length (with 0 for no automatic hard line breaks). Another companion feature would be to reformat a selection. For example, you made modifications to the center of a paragraph and now a line is only half full. This companion feature would readjust all selected lines to maximally use each line by moving the line breaks.

Thanks for considering this!

**Comment by lsawade** [comment]
It's been a long time since this was brought up, but I was surprised it hasn't been implemented! I assume it's not an easy task, but I would like hard-wrapping as well!
Cheers!

**Comment by frentello** [comment]
Yeah the ticket is now almost 5 years up. I would love to see this feature in the hopefully near future

**Comment by beltrame** [comment]
It's very much needed

**Comment by ASalvail** [comment]
For what it's worth, I still think this would be a great addition! I've been doing them manually for all those years, I wouldn't mind a break.

**Comment by rconnorlawson** [comment]
For what it's worth, there's a [TamperMonkey script](https://groups.google.com/g/ace-discuss/c/I6jXl4j8h8k) implementing a paragraph reflow operation like Vim's `gq` command. I haven't tried it myself, but found it on the same day I found this bug. This doesn't address the OP's main request though.

I use vim mode and would love for `gq` to work out of the box!

**Comment by tikacp** [comment]
would be highly appreciated, indeed

**Comment by vbharadwaj-bk** [comment]
Would also appreciate automatic line breaks. I prefer to version-control my LaTeX, and line-level changes are much more meaningful when the line doesn't extend to an entire paragraph.

**Comment by ifigueroap** [comment]
2022 and still no hard breaks nor reflow... Emacs+ AucTex is still the superior editor indeed

**Comment by ckarren** [comment]
Vim mode in Overleaf is hardly usable without hard wraps. If I didn't have to use Overleaf for collabs I would avoid it and stick to vimtex.

**Comment by LarssonOliver** [comment]
I'd also like this feature! Maybe some day...

**Comment by pituach** [comment]
about ten years passed and I upvote this
It is highly needed

**Comment by Jackozee** [comment]
Yes, please. Or even just adding support for the gqq command and related ones.

**Comment by marinegor** [comment]
I gladly join everyone in this thread -- using git/github `diff` tool is so much easier when each sentence is a separate line or few of them. It doesn't break any latex code, but allows you to easily go back to earlier word choice options that you decide you like more.

**Comment by ulysses4ever** [comment]
@ckarren

> Vim mode in Overleaf is hardly usable without hard wraps.

I thought exactly the same until I saw people here mentioning user scripts (e.g. TamperMonkey-based) to customize the shortcuts. E.g. [this script](https://greasyfork.org/en/scripts/461062-overleaf-editor-custom-vim-keybindings-code-mirror-v6) remaps j to gj and k to gk -- and this goes a long way! Of course, 0/$ are not as usable, but still it's a big improvement. Highly recommend!

**Comment by Ahmad-Fermion** [comment]
It is very unfortunate that this feature is still not available!

**Comment by VianaSamuel** [comment]
This feature would be so useful. It's a pity it hasn't been added yet.

**Comment by nonodev96** [comment]
Is there any current way to change the default word wrap?

I've tried to modify with tappermonkey and CodeMirror, it's all very confusing.


```js
(function () {
    'use strict';
    window.addEventListener("UNSTABLE_editor:extensions", (event) => {
        const { CodeMirror, CodeMirrorVim, extensions } = event.detail;
        const { 
          Decoration, EditorSelection, EditorView, Prec, StateEffect, 
          StateField, ViewPlugin, WidgetType, hoverTooltip, keymap, syntaxTree 
        } = CodeMirror

        console.log({ Decoration, EditorSelection, EditorView, Prec, StateEffect });

    });
})();
```

**Comment by cociclo** [comment]
How do people even use things like diff?! But even if we leave our the fact that this is 2025 and people collaborate using things like git that more or less depend on sane lines to be usable, the fact that TeX error messages are line oriented should be reason enough to have hard linebreaks since day one.

This has been an open issue since forever: obviously there is no interest in taking care of it.

**Comment by beltrame** [comment]
Here's a TamperMonkey script that lets you reflow text with Alt-Q like Emacs. It's not the solution but at least solves some headaches.  Ctrl-Alt-F to activate reflow as you type.

https://gist.github.com/beltrame/afe9ec0203f18e2085e7469107754aac

**Comment by ASalvail** [comment]
It must be my birthday today. Thanks @beltrame !
